### PR TITLE
fix(payments-plugin): Calculate tax per orderline instead of per unit for Mollie

### LINKED ISSUE #1939: Cannot checkout with coupons using Mollie payment since 1.9 (Mollie Orders API)
**Describe the bug**
The new Mollie plugin introduced in Vendure 1.9 cannot create payment intents if a coupon is applied due to a VAT mismatch.

@martijnvdbrug 

**To Reproduce**
1. Use the Mollie 1.9 plugin
2. Apply a coupon code to the order, e.g. 50%
3. Attempt to create a Mollie payment intent using the corresponding mutation
4. Creation will fail as the Orders API expects a different VAT than what is supplied

**Expected behavior**
Payment intent creation should work fine and applied coupons should be included the way Mollie expects them to: https://docs.mollie.com/orders/handling-discounts#discounts-and-vouchers

**Environment (please complete the following information):**
 - @vendure/core version: 1.9 and up
 - NodeJS 16
 - Database (mysql/postgres etc): postgres

**Additional context**
Example server output:
```
error 12/17/22, 11:45 PM - [Vendure Server] Order line 1 is invalid. VAT amount is off. Expected VAT amount to be €1.76 (€11.00 × (19.00% / 119.00%)), got €3.51
ApiError: Order line 1 is invalid. VAT amount is off. Expected VAT amount to be €1.76 (€11.00 × (19.00% / 119.00%)), got €3.51
    at Function.createFromResponse (/home/node/app/node_modules/@mollie/api-client/dist/mollie.cjs.js:355:12)
    at throwApiError (/home/node/app/node_modules/@mollie/api-client/dist/mollie.cjs.js:754:22)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)```

### PR DESCRIPTION
**Pull request by martijnvdbrug**
closes #1939

### CODE REVIEW

File: packages/payments-plugin/e2e/mollie-payment.e2e-spec.ts [hunk 1]
```diff
@@ -220,9 +220,11 @@ describe('Mollie payments', () => {
         expect(mollieRequest?.webhookUrl).toEqual(
             `${mockData.host}/payments/mollie/${E2E_DEFAULT_CHANNEL_TOKEN}/1`,
         );
-        expect(mollieRequest?.amount?.value).toBe('2927.60');
-        expect(mollieRequest?.amount?.currency).toBeDefined();
+        expect(mollieRequest?.amount?.value).toBe('1009.90');
+        expect(mollieRequest?.amount?.currency).toBe('USD');
+        expect(mollieRequest.lines[0].vatAmount.value).toEqual('199.98');
         let totalLineAmount = 0;
+        console.log(mollieRequest.lines)
```
Comment (michaelbromley) [inline packages/payments-plugin/e2e/mollie-payment.e2e-spec.ts]: Does this need cleaning up?

File: packages/payments-plugin/src/mollie/mollie.helpers.ts [hunk 2]
```diff
@@ -67,6 +70,15 @@ export function toAmount(value: number, orderCurrency: string): Amount {
     };
 }
 
+/**
+ * Calculate tax amount per order line instead of per unit for Mollie
```
Comment (michaelbromley) [inline packages/payments-plugin/src/mollie/mollie.helpers.ts]: I feel like we need a little mor explanation of the issue this is trying to solve for future reference. A link to issue #1939 and a brief note about why we can't rely on the Vendure lineTax value. Basically the example given in [this comment](https://github.com/vendure-ecommerce/vendure/issues/1939#issuecomment-1362962133) makes clear why we need this work-around.

File: packages/payments-plugin/e2e/mollie-payment.e2e-spec.ts [hunk 3]
```diff
@@ -138,8 +138,8 @@ describe('Mollie payments', () => {
     it('Should prepare an order', async () => {
         await shopClient.asUserWithCredentials(customers[0].emailAddress, 'test');
         const { addItemToOrder } = await shopClient.query<AddItemToOrder.Mutation, AddItemToOrder.Variables>(ADD_ITEM_TO_ORDER, {
-            productVariantId: 'T_1',
-            quantity: 2,
+            productVariantId: '5',
```
Comment (michaelbromley) [inline packages/payments-plugin/e2e/mollie-payment.e2e-spec.ts]: Use the `T_5` form for consistency (even though just the integer form also happens to work).

**Review by michaelbromley — COMMENTED** [review]
Thanks Martijn! Just a couple of small points to review :)

**Comment by martijnvdbrug** [comment]
Yep, fixed! (+ a little fix in the e2e tests regarding refunding)

**Comment by michaelbromley** [comment]
Thanks @martijnvdbrug!

**Pull request merged**